dvc/machine/__init__.py

```diff
@@ -59,9 +64,18 @@ def __init__(
         **kwargs,
     ) -> None:
         selected = selected or list(self.DEFAULT)
-        self.backends = {key: self.DEFAULT[key] for key in selected}
-
-        self.initialized: Dict[str, BaseMachineBackend] = {}
+        self.backends: Dict[str, "BackendCls"] = {}
+        for key in selected:
+            cls = self.DEFAULT.get(key)
+            if cls is None:
+                raise DvcException(
+                    f"'dvc machine' backend '{key}' is missing required "
+                    "dependencies. Install them with:\n"
+                    f"\tpip install dvc[{key}]"
```

Comment:
This will be raised when `feature.machine` is enabled and the `tpi` package is missing. When `feature.machine` is disabled (by default) this code block won't be hit (since backends aren't initialized).